### PR TITLE
[INLONG-6780][Sort] Supports dirty data side-output for JDBC connector and related sinks

### LINKED ISSUE #6780: [Feature][Sort] Supports dirty data side-output for JDBC connector and related sinks
### Description

Description
Supports dirty data side-output for jdbc related connectors.
In this part:

Load 'DirtySinkFactory' and create 'DirtySink' by the config
It is needs to determine whether it is dirty data in the connector.
Side output dirty data by the 'DirtySink' dependents on the configured, the built-in side-out of dirty data has 'LogDirtySink'(https://github.com/apache/inlong/pull/6618) and 'S3DirtySink'(https://github.com/apache/inlong/pull/6655).

### PR DESCRIPTION
**Pull request by Yizhou-Yang**
add jdbc flush exception via batchmap (which is actually a batchset for jdbc)

fix #6780

### CODE REVIEW

File: inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java [hunk 1]
```diff
@@ -91,13 +93,12 @@
...
     private transient volatile Exception flushException;
     private transient RuntimeContext runtimeContext;
-
     private transient ListState<MetricState> metricStateListState;
     private transient MetricState metricState;
     private SinkMetricData sinkMetricData;
     private Long dataSize = 0L;
     private Long rowSize = 0L;
-
+    private final Set<In> batchMap = new HashSet<>();
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java:101]: batchSet

**Comment by dockerzhang** [comment]
@Yizhou-Yang please create a new issue to link this PR, thanks.

File: inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java [hunk 2]
```diff
@@ -320,6 +317,7 @@ public synchronized void flush() throws IOException {
             } catch (SQLException e) {
                 LOG.error("JDBC executeBatch error, retry times = {}", i, e);
                 if (i >= executionOptions.getMaxRetries()) {
+                    clearBatch(e);
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java:320]: If you choose to throw an exception, there is no need to report dirty data

File: inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java [hunk 3]
```diff
@@ -330,6 +328,7 @@ public synchronized void flush() throws IOException {
                     LOG.error(
                             "JDBC connection is not valid, and reestablish connection failed.",
                             exception);
+                    clearBatch(exception);
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java:331]: If you choose to throw an exception, there is no need to report dirty data

**Comment by Yizhou-Yang** [comment]
close for now, I don't think this is necessary to be merged rn

**Pull request closed without merging**